Homework7-start/Homework7/User/UserDetail.swift

```diff
@@ -0,0 +1,197 @@
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import SwiftUI
+
+struct UserDetail: View {
+  let user: User
+  
+  var body: some View {
+    //   VStack {
+//    VStack {
+//      Text("User Info")
+//        .font(.largeTitle)
+//        .fontWeight(.bold)
+//      
+//      //.padding(.bottom, 20)
+//      //     }
+//      
+      
+      NavigationView {
+          VStack {
+            List {
+              Section (header: Text("Personal Information")) {
+                VStack(alignment: .leading) {
+                  HStack {
+                    Text("Name: ")
+                      .font(.headline)
+                    Text("\(user.name.title) \(user.name.first) \(user.name.last)")
+                  }
+                  HStack {
+                    Text("Date of Birth:")
+                      .font(.headline)
+                    Text("\(formattedDate(from: user.dob.date))")
+                    
+                    Text("Age: ")
+                      .font(.headline)
+                    Text("\(user.dob.age)")
+                  }
+                }
+              }
+              Section (header: Text("Location")) {
+                HStack(alignment: .top) {
+                  Text("Address: ")
+                    .font(.headline)
+                  VStack (alignment: .leading){
+                    Text("\( String(user.location.street.number)) \(user.location.street.name) \n\(user.location.city) \(user.location.state) \(String(user.location.postcode))")
+                  }
+                }
+              }
+              Section(header: Text("Coordinates")) {
+                VStack(alignment: .leading){
+                  HStack {
+                    Text("Lattitude: ")
+                      .font(.headline)
+                    Text("\(user.location.coordinates.latitude)")
+                  }
+                  HStack {
+                    Text("Longitude")
+                      .font(.headline)
+                    Text(user.location.coordinates.longitude)
+                  }
+                }
+              }
+              Section (header: Text("Timezone")) {
+                HStack {
+                  Text("Offset: ")
+                    .font(.headline)
+                  Text(user.location.timezone.offset)
+                }
+                HStack {
+                  Text("Description: ")
+                    .font(.headline)
+                  Text(user.location.timezone.description)
+                }
+              }
```

Comment:
Chunks of code that a very similar to one another. You can extract Section, HStack, and the Texts into a view and just pass the section title, title and value instead as parameters of the view and remove all this code that is repeating here.